eslint-local-rules.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview A collection of eslint rules written specifically for
+ * Lighthouse. These are included by the eslint-plugin-local-rules plugin.
+ */
+
+const path = require('path');
+
+/** @typedef {import('eslint').Rule.RuleModule} RuleModule */
+
+/**
+ * Use `require.resolve()` to resolve the location of `path` from a location of
+ * `baseDir` and return it. Returns null if unable to resolve a path.
+ * @param {string} path
+ * @param {string} baseDir
+ * @return {string|null}
+ */
+function requireResolveOrNull(path, baseDir) {
+  try {
+    return require.resolve(path, {
+      paths: [baseDir],
+    });
+  } catch (err) {
+    return null;
+  }
+}
+
+/**
+ * An eslint rule ensuring that any require() of a local path (aka not a core
+ * module or a module dependency) includes a file extension (.js' or '.json').
+ * @type {RuleModule}
+ */
+const requireFileExtension = {
+  meta: {
+    docs: {
+      description: 'disallow require() without a file extension',
```

Comment:
or wait, is this the message that shows up in eslint CLI?

Reply:
I think this is actually if you generate docs for the rule?

On the command line you just get `error  Local require path must have a file extension`

Reply:
Ah gotcha!